### PR TITLE
Added yaml package, yaml utils and resource files

### LINKED ISSUE #87: yaml parser
As a developer, I would like to use .yaml files in cases where I'm initialising a project to a Kubernetes environment.
As a user, I would like to have the ability to create .yaml files easily.

### PR DESCRIPTION
**Pull request by veikkahonkanen**
Resolves #87

### CODE REVIEW

File: src/classes/yaml-utils.ts [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+  /**
+   * Setup service.yaml
+   * 
+   * @param args Object that includes needed values for setting up the .yaml files
+   * @param file The JSON file that is being edited to create a .yaml file
+   */
+  private static setupService = (args: any, file: any): any => {
+    if (args.ports) {
+      file.spec.ports = []
```
Comment (eetupur) [inline src/classes/yaml-utils.ts]:
```suggestion
      file.spec.ports = [];
```

**Review by eetupur — CHANGES_REQUESTED** [review]
found a missing ;

File: resources/deployment.json [hunk 2]
```diff
@@ -0,0 +1,52 @@
+{
```
Comment (jarinykanen) [inline resources/deployment.json:1]: Indentation of this file seems weird

File: resources/service.json [hunk 3]
```diff
@@ -0,0 +1,23 @@
+{
```
Comment (jarinykanen) [inline resources/service.json:1]: Indentation of this file looks weird

File: src/classes/yaml-utils.ts [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+   * CREATE
+   * Creates .yaml files that are used when putting up a project into Kubernetes Minikube
+   * 
+   * @param args Object that includes needed values for setting up the .yaml files
+   * @param type Type of Kubernetes component, either pod, service or deployment
+   * @param repoPath Repository path where to init .yaml files
+   */
+  public static createYaml = async (args: any, type: string, repoPath: string): Promise<void> => {
+
+    let file : any = fs.readFileSync(`./resources/${type}.json`);
```
Comment (jarinykanen) [inline src/classes/yaml-utils.ts]:
```suggestion
    let file: any = fs.readFileSync(`./resources/${type}.json`);
```

File: src/classes/yaml-utils.ts [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+    file.metadata.name = args.name;
+
+    if (file.metadata.labels) {
+
+      if (file.metadata.labels.app) file.metadata.labels.app = args.name;
+      if (file.metadata.labels.run) file.metadata.labels.run = args.name;
```
Comment (jarinykanen) [inline src/classes/yaml-utils.ts]:
```suggestion
      if (file.metadata.labels.app) {
        file.metadata.labels.app = args.name;
      }

      if (file.metadata.labels.run) {
        file.metadata.labels.run = args.name;
      }
```

File: src/classes/yaml-utils.ts [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+
+  /**
+   * READ
+   * Prints out an existing .yaml file
+   * 
+   * @param type Type of Kubernetes component, either pod, service or deployment
+   * @param repoPath Repository path where .yaml files are located
+   */
+  public static printYaml = async (type: string, repoPath: string): Promise<void> => {
+
```
Comment (jarinykanen) [inline src/classes/yaml-utils.ts]:
```suggestion
```

File: src/classes/yaml-utils.ts [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+  /**
+   * CREATE
+   * Creates .yaml files that are used when putting up a project into Kubernetes Minikube
+   * 
+   * @param args Object that includes needed values for setting up the .yaml files
+   * @param type Type of Kubernetes component, either pod, service or deployment
+   * @param repoPath Repository path where to init .yaml files
+   */
+  public static createYaml = async (args: any, type: string, repoPath: string): Promise<void> => {
+
```
Comment (jarinykanen) [inline src/classes/yaml-utils.ts]:
```suggestion
```

File: src/classes/yaml-utils.ts [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+  /**
+   * Setup pod.yaml
+   * 
+   * @param args Object that includes needed values for setting up the .yaml files
+   * @param file The JSON file that is being edited to create a .yaml file
+   */
+  private static setupPod = (args: any, file: any): any => {
+    file.spec.containers[0].name = args.name;
+    if (args.image) file.spec.containers[0].image = args.image;
+    if (args.port) file.spec.containers[0].ports[0].containerport = args.port;
```
Comment (jarinykanen) [inline src/classes/yaml-utils.ts]:
```suggestion
    if (args.image) {
      file.spec.containers[0].image = args.image;
    }

    if (args.port) {
      file.spec.containers[0].ports[0].containerport = args.port;
    }
```

File: src/classes/yaml-utils.ts [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+  }
+
+  /**
+   * Setup pod.yaml
+   * 
+   * @param args Object that includes needed values for setting up the .yaml files
+   * @param file The JSON file that is being edited to create a .yaml file
+   */
+  private static setupPod = (args: any, file: any): any => {
+    file.spec.containers[0].name = args.name;
```
Comment (jarinykanen) [inline src/classes/yaml-utils.ts]: Add undefined checks for container[0] and container[0].ports[0]

File: src/classes/yaml-utils.ts [hunk 10]
```diff
@@ -0,0 +1,126 @@
...
+  /**
+   * Setup service.yaml
+   * 
+   * @param args Object that includes needed values for setting up the .yaml files
+   * @param file The JSON file that is being edited to create a .yaml file
+   */
+  private static setupService = (args: any, file: any): any => {
+    if (args.ports) {
+      file.spec.ports = [];
+      for (let port in args.ports) file.spec.ports.push(port);
```
Comment (jarinykanen) [inline src/classes/yaml-utils.ts]:
```suggestion
      for (let port in args.ports) {
        file.spec.ports.push(port);
      }
```

File: src/classes/yaml-utils.ts [hunk 11]
```diff
@@ -0,0 +1,126 @@
...
+    }
+    file.spec.selector.app = args.name;
+    if (args.portType) file.spec.type = args.portType;
```
Comment (jarinykanen) [inline src/classes/yaml-utils.ts]:
```suggestion
    if (args.portType) {
      file.spec.type = args.portType;
    }
```

File: src/classes/yaml-utils.ts [hunk 12]
```diff
@@ -0,0 +1,126 @@
...
+    if (args.ports) {
+      file.spec.template.spec.containers[0].ports = []
+      for (let port in args.ports) file.spec.template.spec.containers[0].ports.push(port);
+    }
+    if (args.env) {
+      for (let env in args.env) file.spec.template.spec.containers[1].env.push(env);
+    }
+    if (args.replicas) file.spec.replicas = args.replicas;
+    return file;
+  }
```
Comment (jarinykanen) [inline src/classes/yaml-utils.ts]: Check comments from above

**Review by jarinykanen — CHANGES_REQUESTED** [review]

File: src/classes/yaml-utils.ts (hunk 12 above)
Comment (veikkahonkanen) [reply src/classes/yaml-utils.ts]: Switched to using curly brackets

File: src/classes/yaml-utils.ts (hunk 9 above)
Comment (veikkahonkanen) [reply src/classes/yaml-utils.ts]: Added undefined checks

File: resources/deployment.json (hunk 2 above)
Comment (veikkahonkanen) [reply resources/deployment.json:1]: Renewed indentation of the file

File: resources/service.json (hunk 3 above)
Comment (veikkahonkanen) [reply resources/service.json:1]: Renewed indentation of the file

**Review by eetupur — APPROVED** [review]

File: src/classes/yaml-utils.ts [hunk 13]
```diff
@@ -0,0 +1,154 @@
...
+
+  /**
+   * Setup deployment.yaml
+   * 
+   * @param args Object that includes needed values for setting up the .yaml files
+   * @param file The JSON file that is being edited to create a .yaml file
+   */
+  private static setupDeployment = (args: any, file: any): any => {
```
Comment (jarinykanen) [inline src/classes/yaml-utils.ts:129]: Add return docs

File: src/classes/yaml-utils.ts [hunk 14]
```diff
@@ -0,0 +1,154 @@
...
+    file.spec.selector.app = args.name;
+    file.spec.template.metadata.app = args.name;
+
+    if (file.spec.template.spec.containers[0]) {
```
Comment (jarinykanen) [inline src/classes/yaml-utils.ts]: Take file.spec.template.spec.containers[0] to own variable --> This will make this function much easier to read

**Review by jarinykanen — CHANGES_REQUESTED** [review]

File: src/classes/yaml-utils.ts (hunk 13 above)
Comment (veikkahonkanen) [reply src/classes/yaml-utils.ts:129]: added return docs

File: src/classes/yaml-utils.ts (hunk 14 above)
Comment (veikkahonkanen) [reply src/classes/yaml-utils.ts]: taken to own variable

File: src/classes/yaml-utils.ts [hunk 15]
```diff
@@ -0,0 +1,158 @@
...
+
+  /**
+   * Setup pod.yaml
+   * 
+   * @param args Object that includes needed values for setting up the .yaml files
+   * @param file The JSON file that is being edited to create a .yaml file
+   * @returns the edited JSON file for pod
+   */
+  private static setupPod = (args: any, file: any): any => {
+    if (file.spec.containers[0]){
```
Comment (jarinykanen) [inline src/classes/yaml-utils.ts]: Same thing here as in setupDeployment (own variable)

**Review by jarinykanen — CHANGES_REQUESTED** [review]

**Review by eetupur — APPROVED** [review]

File: src/classes/yaml-utils.ts (hunk 15 above)
Comment (veikkahonkanen) [reply src/classes/yaml-utils.ts]: containers[0] has been assigned to a variable

**Review by eetupur — APPROVED** [review]

**Review by jarinykanen — APPROVED** [review]

**Pull request merged**